tsconfig.json

```diff
@@ -6,20 +6,16 @@
     "module": "commonjs",
     "pretty": true,
     "noEmitOnError": true,
-    "experimentalDecorators": true,
-    "sourceMap": true,
-    "emitDecoratorMetadata": true,
     "strict": true,
     "noImplicitAny": false,
-    "esModuleInterop": true,
```

Comment:
What makes this harmful?

Reply:
Because if a library is compiled using `esModuleInterop: true` if the final user has `skipLibCheck: false` (default value) TS will try to check the lib typyings too. If `esModuleInterop` is set to `false` it will not compile. Showing the error about imports. On the other hand using this like this will make the types work in every condition. We had this issues in past all over a place. I think the best we can do is to make tests that use every possible combination of this values.

Reply:
Ah, ok, that makes sense, it has to be the most conservative syntax so it works with all the consuming `esModuleInterop` settings.